runtime-attach/src/main/java/io/opentelemetry/contrib/attach/RuntimeAttach.java

```diff
@@ -15,8 +15,15 @@ public static void attachJavaagentToCurrentJVM() {
     if (agentIsDisabled()) {
       return;
     }
-    File javaagentFile = AgentFileLocator.locateAgentFile();
-    ByteBuddyAgent.attach(javaagentFile, getPid());
+
+    AgentFileProvider agentFileProvider = new AgentFileProvider();
+    File javaagentFile = agentFileProvider.getAgentFile();
+
+    try {
+      ByteBuddyAgent.attach(javaagentFile, getPid());
+    } finally {
+      agentFileProvider.deleteTempDir();
```

Comment:
I would think that the JVM would still need access to the jar file for lazy loading additional classes? (in which case we could delete on exit instead)

Reply:
Not sure because the Javadoc of Byte Buddy's `attach(File agentJar, String processId)` method says _"This operation blocks until the attachment is complete"_.  In case of doubt, I will delete in exit.